public/graph.js

```diff
@@ -1,251 +1,181 @@
-
-        let offsetWeek = new Date(),
-            offsetMonth = new Date();
-
-        offsetWeek.setDate(offsetWeek.getDate() - TIMEFRAME_WEEK);
-        offsetMonth.setDate(offsetMonth.getDate() - TIMEFRAME_MONTH);
-
-        // Set default y-axis limits
-        let dataFilteredWeek = data.filter(a => a.datetime > offsetWeek),
-            dataFilteredMonth = data.filter(a => a.datetime > offsetMonth);
-
-        // Group received data by day
-        let dailyReceivedTotal = d3
+    static GroupDataByDay(dataFilteredMonth, messageDirection) {
+        let groupedDataTotal = d3
             .nest()
             .key(d => d.day)
             .rollup(v => {
-                let receivedData = {};
-                operators.forEach(operator => {
-                    receivedData[`${operator}_received`] = d3.sum(
-                        v,
-                        d => d[`${operator}_received`]
+                let groupedData = {};
+                GC.operators.forEach(operator => {
+                    groupedData[`${operator}_${messageDirection}`] = d3.sum(v,
+                        d => d[`${operator}_${messageDirection}`]
                     );
                 });
-                receivedData["total_received"] = d3.sum(v, d => d.total_received);
-                return receivedData;
+                groupedData[`total_${messageDirection}`] = d3.sum(v,d => d[`total_${messageDirection}`]
+                );
+                return groupedData;
             })
             .entries(dataFilteredMonth);
-
-        // Flatten nested data for stacking
-        for (let entry in dailyReceivedTotal) {
-            let valueList = dailyReceivedTotal[entry].value;
-            for (let key in valueList) {
-                dailyReceivedTotal[entry][key] = valueList[key];
-            }
-            dailyReceivedTotal[entry]["day"] = dailyReceivedTotal[entry].key;
-            delete dailyReceivedTotal[entry]["value"];
-            delete dailyReceivedTotal[entry]["key"];
+        if (messageDirection == "sent") {
+            GC.dailySentTotal = groupedDataTotal;
+        } else if (messageDirection == "received") {
+            GC.dailyReceivedTotal = groupedDataTotal;
         }
+    }
 
-        // Group sent data by day
-        let dailySentTotal = d3
-            .nest()
-            .key(d => d.day)
-            .rollup(v => {
-                let sentData = {};
-                operators.forEach(operator => {
-                    sentData[`${operator}_sent`] = d3.sum(v, d => d[`${operator}_sent`]);
-                });
-                sentData["total_sent"] = d3.sum(v, d => d.total_sent);
-                return sentData;
-            })
-            .entries(dataFilteredMonth);
-
-        // Flatten nested data for stacking
-        for (let entry in dailySentTotal) {
-            let valueList = dailySentTotal[entry].value;
+    static FlattenNestedDataforStacking(messageDirection) {
```

Comment:
```suggestion
    static flattenNestedDataforStacking(messageDirection) {
```
because this is a method